### DISCUSSION TITLE
Specifications on [START] and [END] timestamps?

### DISCUSSION THREAD

**Discussion by krooonal** [ambros-gleixner/MIPcc23#16] [Q&A] [answered]
Hi,
In the python example, the start time is printed just before the optimize call. The end time is printed right after that. The following parts are outside of the reported timestamps:
- Problem loading (reading the file and putting it in the solver)
- Solver configurations (parameter setting of the solver)
- Solution reporting (writing the solution file)
This makes me wonder, what else are we allowed to do outside of those timestamps? Clearly, reusing solve information requires storing and loading (and sometimes manipulating) from the solver. Can this also be done outside of the start and end time stamps?

Looking at the evaluation code, it seems to penalize the solution reporting after end timestamp. However, there is nothing to check the start timestamp. Will that be checked manually?

**Comment by matbesancon** [comment, marked as answer]
Hi!
All code bases will be checked manually to ensure that no computation is done outside of the time stamp.
As a rule of thumb, reading a file and setting a parameter would be okay, computing anything would not (even something as trivial as sorting an array must be counted within the time stamp bounds).

Does that answer your question?

**Reply by krooonal** [reply]
Yes, thanks.

A follow-up question: Is it advisable to use 1 or 2 seconds less than the specified time limit for solving (and other computations) to make sure that the solutions file writing part is done within the limit? I would do this for safety, but it can also get me in trouble if there are private series with very small time limit (5-10 seconds).

**Reply by matbesancon** [reply]
The file writing part should not be an issue in any case no. If we detect a warning because of a mismatch on the times, we will check the file writing time and the code itself to be sure this is just writing.
